#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *
PACKAGE = "camera"

gen = ParameterGenerator()


awb_mode_enum = gen.enum([gen.const("Off", str_t, "off", "Turn off auto white balance"),
                          gen.const("Auto", str_t, "auto", "Set white balance to automatic"),
                          gen.const("Sunlight", str_t, "sunlight", "Set white balance to sunlight"),
                          gen.const("Cloudy", str_t, "cloudy", "Set white balance to cloudy"),
                          gen.const("Shade", str_t, "shade", "Set white balance to shade"),
                          gen.const("Tungsten", str_t, "tungsten", "Set white balance to tungsten"),
                          gen.const("Fluorescent", str_t, "fluorescent", "Set white balance to fluorescent"),
                          gen.const("Incandescent", str_t, "incandescent", "Set white balance to incandescent"),
                          gen.const("Flash", str_t, "flash", "Set white balance to flash"),
                          gen.const("Horizon", str_t, "horizon", "Set white balance to horizon"),
                          ], "Enumeration to set white balance")

exposure_mode_enum = gen.enum([gen.const("Off", str_t, "off", "Turn off automatic exposure"),
                               gen.const("Auto", str_t, "auto", "Set exposure mode to auomatic"),
                               gen.const("Night", str_t, "night", "Set exposure mode to night"),
                               gen.const("NightPreview", str_t, "nightpreview", "Set exposure mode to night preview"),
                               gen.const("Backlight", str_t, "backlight", "Set exposure mode to backlight"),
                               gen.const("Spotlight", str_t, "spotlight", "Set exposure mode to spotlight"),
                               gen.const("Sports", str_t, "sports", "Set exposure mode to sports"),
                               gen.const("Snow", str_t, "snow", "Set exposure mode to snow"),
                               gen.const("Beach", str_t, "beach", "Set exposure mode to beach"),
                               gen.const("VeryLong", str_t, "verylong", "Set exposure mode to very long"),
                               gen.const("FixedFPS", str_t, "fixedfps", "Set exposure mode to fixed fps"),
                               gen.const("AntiShake", str_t, "antishake", "Set exposure mode to anti shake"),
                               gen.const("Fireworks", str_t, "fireworks", "Set exposure mode to fireworks"),
                               ], "Enumeration so set exposure mode")

gen.add("awb_mode", str_t, 0, "AWB Parameter", "off", edit_method=awb_mode_enum)
gen.add("shutter_speed", int_t, 0, "Shutter Speed in micro seconds. Default 0 (auto)", 0, 0, 100)
gen.add("exposure_mode", str_t, 0, "Exposure Mode", "auto", edit_method=exposure_mode_enum)
gen.add("brightness", int_t, 0, "Brightness", 50, 0, 100)
gen.add("iso", int_t, 0, "Default 0 (auto)", 0, 0, 1600)
gen.add("awb_gain_red", double_t, 0, "Red gain", 1.0, 0.0, 8.0)
gen.add("awb_gain_blue", double_t, 0, "Blue gain", 1.0, 0.0, 8.0)

exit(gen.generate(PACKAGE, "camera_node", "cam_params"))

